### PR TITLE
implemented cache decorator

### PR DESCRIPTION
**Pull request by BGBroth**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
-    # Write your code here
-    pass
+from typing import Callable
+
+
+cache_dict = {}
+
+
+def cache(func: Callable) -> Callable:
+    def inner(*args) -> int:
```
Comment (Polyakiv-Andrey) [inline app/main.py:8]: you lose the metadata of the function being wrapped (its name, etc.)

**Review by Polyakiv-Andrey — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+
+
+cache_dict = {}
+
+
+def cache(func: Callable) -> Callable:
+    @wraps(func)
+    def wrapper(*args: Any) -> int:
+        if func not in cache_dict:
+            cache_dict[func] = {}
```
Comment (Abnormaltype) [inline app/main.py:12]: unnecessary if, you don't need it

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
+        if args not in cache_dict[func]:
+            print("Calculating new result")
+            cache_dict[func][args] = func(*args)
+        else:
```
Comment (Abnormaltype) [inline app/main.py:16]: unnecessary `else`

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Review by Polyakiv-Andrey — APPROVED** [review]
Good job!

**Pull request closed without merging**